proposals/xxxx-wave-matrix.md

```diff
@@ -0,0 +1,396 @@
+number of elements per row. Any value below the minimum legal values are
+ignored. The behavior of row stride values that are not a multiple of element
+stride is undefined.
+
+#### WaveMatrix Left & Right
+
+```c++
+template <typename ElTy, int NRows, int NCols>
+class WaveMatrixLeft : detail::WaveMatrixBase<ElTy, NRows, NCols> {
+  uint MatrixDepth();
+};
+
+template <typename ElTy, int NRows, int NCols>
+class WaveMatrixRight : detail::WaveMatrixBase<ElTy, NRows, NCols> {
+  uint MatrixDepth();
+};
+```
+
+`ElTy` must be either a 32 or 16 bit floating point type or an 8 bit packed
+signed or unsigned integer type. `NRows` and `NCols` must be compile-time
+constant expressions.
+ 
+##### WaveMatrix(Left|Right) MatrixDepth
+
+The `MatrixDepth` method returns the hardware-dependent depth for the matrix
+multiplication unit. The resulting value must be an even multiple of 16.
+
+### WaveMatrix Matrix Fragment Objects
+
+The code below approximately declares the base interface that all WaveMatrix
+fragment objects implement. A WaveMatrix fragment object stores a single row or
+column of a WaveMatrix matrix.
+
+```c++
+namespace detail {
+template <typename ElTy, int NRows, int NCols>
+class WaveMatrixFragmentBase : WaveMatrixBase<ElTy, NRows, NCols> {
+  void Fill(ElTy Val);
+
+  void Load(ByteAddressBuffer Res, uint StartOffset, uint Stride,
+            uint Align = 0);
+  void Load(RWByteAddressBuffer Res, uint StartOffset, uint Stride,
+            uint Align = 0);
+
+  void Load(groupshared ElTy Arr[], uint StartIdx, uint Stride);
+
+  void Store(RWByteAddressBuffer Res, uint StartOffset, uint Stride,
+             uint Align = 0);
+
+  void Store(groupshared ElTy Arr[], uint StartIdx, uint Stride);
+};
+} // namespace detail
+```
+
+#### Loading and Storing WaveMatrix Fragment Objects
+
+Values for WaveMatrix fragment objects can be loaded from and stored to
+`[RW]ByteAddressBuffer` or `groupshared` array of type `ElTy`.
+
+Loading from or storing to a `[RW]ByteAddressBuffer` takes a start offset in
+bytes, element stride in bytes, and optional alignment in bytes. The alignment
+must be at least `sizeof(ElTy)` otherwise the load behavior is undefined.
+
+Loading from or storing to a `groupshared` array takes the starting index, and
+the element stride as a number of elements.
+
+##### Stride
+
+When loading and storing from `groupshared` arrays, the stride is expressed in
+number of elements.
+
+When loading and storing from `[RW]ByteAddressBuffer` types the stride must be
+greater than or equal to the size of the element type. Any value below the
+minimum legal values are ignored. The behavior of row stride values that are not
+a multiple of element stride is undefined.
+
+### WaveMatrix Accumulator Objects
+
+The WaveMatrix Accumulator objects come in three forms which are represented by
+two categories: matrix accumulators and fragment accumulators. All accumulators

```

Comment:
I found "which are represented by" to be confusing because there are 3 of one and 2 of another. Would "fall into two categories" be equally accurate?